Libraries/Utilities/Appearance.js

```diff
@@ -46,7 +48,11 @@ if (NativeAppearance) {
           colorScheme == null,
         "Unrecognized color scheme. Did you mean 'dark' or 'light'?",
       );
-      eventEmitter.emit('change', {colorScheme});
+
+      if (previousPreferences?.colorScheme !== colorScheme) {
+        previousPreferences = newAppearance;
+        eventEmitter.emit('change', {colorScheme});
+      }
```

Comment:
thanks for looking into the issue.

i'm just little afraid these code changes in JS side would not apply to expo projects. previously, users will [have our fork in dependencies](https://github.com/expo/expo/blob/b643b5f59cbb98fed84901e54b985f57bec0ea97/templates/expo-template-blank/package.json#L18). after sdk43, we [use the official react-native](https://github.com/expo/expo/blob/345a2db8b6f1c4fe23e2f2773ff4226566665f35/templates/expo-template-blank/package.json#L18). so we can only change native side in expo go.

do you think whether it's possible we can do it only in native code?